### ISSUE TITLE
Compiler should not produce warning for nullable type parameter in XML doc cref

### ISSUE THREAD

**Issue by ElStefan** [dotnet/roslyn#67737]
**Version Used**: 
VS 17.5.3, C#11

**Steps to Reproduce**:

See [SharpLab](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQODEAdgK4QQCGARhAKYAE1BlNOOyAzLWAcNQE4BmZAMZ0AkgBVqAZ2A4A3jlpLaAejW0APFKIBbHWV4BPAHyLlalbQDKAex10eAD2AA6M0ouaV2vQZPvadkCAFlpJGQBZamAACxsYDXFjAApxAH5aGwoAKwBKAG4cAF8WbCC0MOlgWhBaCSr5AM8NLhi+MGAYGyFaIV5qfgBeACJ6mRdw4CjY+LlxItS03OHVU2xlQI5kUMnpuISkxcyc3ICFdeUS7CugA)
```csharp
using System;

#nullable enable

public interface ITest
{
    /// <summary>
    /// Some text.
    /// </summary>
    public void TestMethod<T>(T? obj);
}

public class Test : ITest
{
    /// <inheritdoc cref="ITest.TestMethod{T}(T?)" />
    public void TestMethod<T>(T? obj)
    {
    }
}
```

**Expected Behavior**:
No warnings
**Actual Behavior**:
A compiler warning CS1574 appears, stating that the given reference method in `cref` cannot be found. 
But navigating to the interface works and hovering with the mouse even shows a tooltip with the correct summary.

**Comment by vbreuss** [comment]
- This issue could have a similar root cause as in #46770:
   Both deal with compiler warnings when using `cref` attributes with nullable types...

**Comment by jcouv** [comment]
I'll have to check what are the rules for nullable annotations in xml docs, but from quick experimentation I noticed that removing the `?` annotation in the xml doc solves the problem ([sharplab](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQODEAdgK4QQCGARhAKYAE1BlNOOyAzLWAcNQE4BmZAMZ0AkgBVqAZ2A4A3jlpLaAejW0APFKIBbHWV4BPAHyLlalbQDKAex10eAD2AA6M0ouaV2vQZPvadkCAFlpJGQBZamAACxsYDXFjAApxAH5aGwoAKwBKAG4cAF8WbCC0MOlgWhBaCSr5AM8NLhi+MGAYGyFaIV5qfgBeACJ6mRdw4CjY+LlxItTc4dVTbGVAjmRQyem4hKTUjKy8gIU15RLsS6A==)).
